remoteworkitem/flatten_test.go

```diff
@@ -0,0 +1,34 @@
+package remoteworkitem
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFlattenMap(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	testString := []byte(`{"name":"shoubhik", "assignee":{"fixes": 2, "complete" : true,"foo":[1,2,3,4],"1":"sbose","2":"pranav","participants":{"4":"sbose56","5":"sbose78"}},"name":"shoubhik"}`)
+	var nestedMap map[string]interface{}
+	err := json.Unmarshal(testString, &nestedMap)
+
+	if err != nil {
+		t.Error("Incorrect dataset ", testString)
+	}
+	OneLevelMap := Flatten(nestedMap)
+
+	// Test for string
+	assert.Equal(t, OneLevelMap["assignee.participants.4"], "sbose56", "Incorrect mapping found for assignee.participants.4")
+
+	// test for int
+	assert.Equal(t, int(OneLevelMap["assignee.fixes"].(float64)), 2)
+
+	// test for array - TODO: Need a better way to handle the automatic conversion of int to float64 during implicit unmarshal )
+	refArray := []interface{}{float64(1), float64(2), float64(3), float64(4)}
+	assert.Equal(t, OneLevelMap["assignee.foo"], refArray)
+
```

Comment:
Missing test scenario to cover array of objects: "val": [ {"a":"b"} ]

Should probably have key; val.0,.1